src/hooks/useNoInternetToast.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { createContext, useEffect, useState } from 'react';
+import { useNetInfo } from '@react-native-community/netinfo';
+import { t } from 'i18next';
+import Toast, { ToastProps } from 'react-native-toast-message';
+
+const NoInternetToastContext = createContext<{
+  refreshToast?: () => void;
+}>({});
+
+export const NoInternetToastProvider = ({
+  children,
+  toastOverrides,
+}: {
+  children?: React.ReactNode;
+  toastOverrides?: ToastProps;
+}) => {
+  const { isConnected } = useNetInfo();
+  const [forceRefresh, setForceRefresh] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (isConnected === false) {
+      Toast.show({
+        type: 'error',
+        text1: t('connection-error-title', 'No internet connection ❌'),
+        text2: t(
+          'connection-error-body',
+          'This app requires an internet connection.',
+        ),
+        autoHide: false,
+        ...toastOverrides,
+      });
+    } else if (isConnected === true) {
+      Toast.hide();
+    }
+  }, [isConnected, forceRefresh, toastOverrides]);
+
+  return (
+    <NoInternetToastContext.Provider
+      value={{
+        refreshToast: () => setForceRefresh((currentVal) => !currentVal),
```

Comment:
You mentioned in the PR desc this will be used to put the toast back up upon user interaction, but can you explain more how that will work?  I wonder about instead experimenting with `visibilityTime` (e.g. 10 min), `autoHide=false`, `onPress` (e.g. can we cancel default behavior in there?), and maybe best of all `onHide` - maybe `onHide` could internally toggle setForceRefresh and then we wouldn't have to expose `refreshToast`?

Reply:
Done!